contracts/governance/src/collection.rs

```diff
@@ -0,0 +1,43 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use alloc::vec::Vec;
+use canonical::CanonError;
+use canonical_derive::Canon;
+
+#[derive(Clone, Canon, Debug)]
+pub struct Collection<K, V> {
+    data: Vec<(K, V)>,
+}
+
+impl<K: PartialEq, V: PartialEq> Collection<K, V> {
+    // Methods return a result to keep things consistent with the map methods
+    pub fn get(&self, key: &K) -> Result<Option<&V>, CanonError> {
+        Ok(self.data.iter().find(|(x, _)| x == key).map(|(_, y)| y))
+    }
+
+    pub fn insert(&mut self, key: K, value: V) -> Result<(), CanonError> {
+        if let Some(pos) = self.data.iter().position(|(k, _)| k == &key) {
+            self.data[pos] = (key, value)
+        } else {
+            self.data.push((key, value))
+        }
+
+        Ok(())
+    }
+
+    pub fn remove(&mut self, key: &K) -> Result<(), CanonError> {
+        self.data.retain(|(k, _)| k == key);
+
+        Ok(())
+    }
+}
+
+impl<K, V> Default for Collection<K, V> {
```

Comment:
We can just derive `Default` for this

Reply:
It doesn't compile when you add the default trait via derive macro

Reply:
Hmm that's odd. `Vec` doesn't implement default there? Anyway, shouldn't be a blocker for this PR